### DISCUSSION TITLE
Run process in the background example

### DISCUSSION THREAD

**Discussion by DevOpsCraftsman** [bigskysoftware/htmx#867] [Ideas]
Hi! Evryone!

I know there is already a progress bar example for background process feedback visualization.  
But I think a more simple example could be added.

My proposition:

```html
<button class="btn" hx-post="/pre_run" hx-swap="outerHTML">
  Run
</button>
```

The `pre_run` endpoint does not yet run the process, but just returns a new button with the load trigger that actually launches it on the server and wait for its completion:

```html
<button class="btn" hx-post="/run" hx-trigger="load" hx-swap="outerHTML">
  Running…
</button>
```

After completion, the response returns a simple button telling we’re done:

```html
<button class="btn">
  Success!
</button>
```

Of course, if there is failure we can returns the first button with another content like "Error… Please retry".

What do you think?

**Comment by DevOpsCraftsman** [comment]
Answering myself :

It's already handled by this in a lot more elegant way:

https://htmx.org/attributes/hx-indicator/

**Reply by ivangretsky** [reply]
If the question is answered, would you mind marking it as such?
Or we can't do this with ideas?

**Reply by DevOpsCraftsman** [reply]
I want to, but didn't find how... Where I'm supposed to click? 
(I fear I can't, because my own messages are considered as comments, not answers...)

**Reply by ivangretsky** [reply]
Sorry. I am new to github discussions. And it seems like [we cannot](https://github.com/github/feedback/discussions/3097) close or mark the idea discussion as resolved. My bad.

**Reply by 78wesley** [reply]
https://github.blog/changelog/2023-03-14-adding-the-ability-to-close-github-discussions/

**Reply by ivangretsky** [reply]
@DevOpsCraftsman does the new option suggested by @78wesley work for you?

**Reply by DevOpsCraftsman** [reply]
Done, thanks!
